README.md

```diff
@@ -32,84 +40,183 @@ spec:
         "bridge": "br1"
       },
       {
-        "type": "cidr-filtering-cni"
+        "type": "bridge-filtering"
       }
     ]
   }'
+```
+
+The `example-network` specifies two CNI plugins, the first cni in the above is a `cnv-bridge`, connecting pods to linux-bridge `br1`, and the second
+is the `bridge-filtering`, that handles configuration of nftables rules on the pod.
+
+By default, all ingress and egress traffic is dropped.
+To allow specific ingress and egress CIDR blocks/ports, create ConfigMaps(s) referencing a NetworkAttachmentDefinition spec.config name in the ConfigMap's label.
+Each ConfigMap additionally needs to have `bridge-filtering` label to be enabled for the bridge-filtering plugin.
+
+The following manifest is an example of a minimal `ConfigMap`, that doesn't yet allow any traffic. Since denying all traffic is done by default, it doesn't needed to exist.
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: deny-all
+  labels:
+    bridge-filtering: ""
+    br1-with-cidr-filtering: ""
+data:
+  config.json: |
+    {
+      "ingress": {},
+      "egress": {}
+    }
+```
+
+Notice the labels of the ConfigMap.
+The first label, `bridge-filtering` ensures that the `ConfigMap` configuration is collected by the CNI plugin, as
+the plugin only lists `ConfigMaps` with this label.
+The second label, `br1-with-cidr-filtering` refers to the `NetworkAttachmentDefinition` spec name, in the CNI configuration JSON.
+
+
+> **_NOTE:_**  A ConfigMap may refer to many NetworkAttachmentDefinition specs. The order in which ConfigMaps are process is not defined.
+
+
+To allow a pod to communicate to external entities, let's create the following `ConfigMap`:
+```yaml
 ---
 kind: ConfigMap
 apiVersion: v1
 metadata:
-  name: local-network
+  name: egress-local-network
   namespace: default
   labels:
-    cidr-filtering-cni: ""
+    bridge-filtering: ""
     br1-with-cidr-filtering: ""
 data:
   config.json: |
     {
-      "ingress": {
+      "egress": {
         "subnets": [
           {
             "subnet": {
-              "cidr": "192.168.66.0/24",
+              "cidr": "192.168.0.0/16",
               "except": [
-                "192.168.66.105"
+                "192.168.150.0/24",
+                "192.168.151.151"
               ]
             }
           }
+        ],
+        "ports": [
+          {
+            "protocol": "TCP",
+            "port": "80"
+          },
+          {
+            "protocol": "tcp",
+            "port": "8080"
+          }
         ]
-      },
-      "egress": {
+      }
+    }
+```
+
+The `egress.subnets` attribute allows pod to reach any IP address in subnet `192.168.0.0/16`, except for subnet `192.168.150.0/24` and `192.168.151.151`.
+The `egress.ports` attribute allows pod to reach ports 80, and 8080 over TCP protocol.
+
+To allow any webserver responses to reach the pod, create the following `ConfigMap`:
```

Comment:
```suggestion
To allow any pod requests to reach the webserver, create the following `ConfigMap`:
```

No?

Reply:
Actually, no. In both, ingress and egress we match destination port of a packet.
The problem is, client opens connection with a service from a random port, to which is then sent the response.
Normally, this is handled by accepting packets matching "related connection". But the nftable module that enables this feature is not installed by default. 
So for ingress, I need to open all tcp ports to have the response accepted.

Alternatively, we could match source port of ingressing packets, which would probably make more sense without conntrack, but for the price that we deviate from NetworkPolicy
/cc @maiqueb

Reply:
yeah, afaiu, we don't do conntrack. We probably should.

I actually wasn't aware we required a particular module to have access to it. Could you add more info about this @rhrazdil ?

Reply:
Another thing that pops to mind: wouldn't the user be expected to provision both ingress / egress in the client to allow the response to go through ? 

*If* so, I think we should document this limitation / requirement we make of them - i.e. dear user, we do not have conntrack. Plan your policies with this in mind.

**If** we eventually decide to implement conntrack, that would change.

Reply:
+1

@rhrazdil let's first evaluate how much extra effort it would require. If it is more than trivial, I'd rather release with the limitation for now

Reply:
Using conntrack shoudln't imply a big change to provisioning the rules, I'll just need to figure out why I wasn't able to create rules that use conntrack (maybe load the kernel module?). But, Miguel found that Calico is using conntrack in nftable rules on our nodes and confirmed that conntrack is available on openshift nodes. So if calico can use it, we must be able to use it as well.

For this PR, I would go with documenting the limitation, so that we can proceed. Then get in other open PRs, especially https://github.com/rhrazdil/cidr-filtering-cni/pull/14, and then deliver conntrack.
Is this an acceptable approach?

Reply:
Works for me, let's get this merged o/